packages/application-shell/src/components/application-helmet/application-helmet.tsx

```diff
@@ -0,0 +1,113 @@
+import type { TEnhancedLocation } from '@commercetools-frontend/browser-history';
+
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+import { Helmet } from 'react-helmet';
+
+type Breadcrumb<Query extends {} = {}> = {
+  suffix: string;
+  paths: string[];
+  location: TEnhancedLocation<Query>;
+};
+type Props<Query extends {} = {}> = {
+  // Allow to render a custom page title.
+  renderPageTitle?: (breadcrumb: Breadcrumb<Query>) => string;
+  children: React.ReactNode;
+};
+
+const maxTitleCharLength = 60;
+
+const compilePath = (paths: string[], hasTruncatedPaths = false): string[] => {
+  if (hasTruncatedPaths) {
+    return [...paths.slice(0, 1), '...', ...paths.slice(1)];
```

Comment:
Love this line ❤️ "Code is poetry!"
LGTM. lets ship it 🚀 Happy 10 years 🥳 🥳 🥳 💯